tests/support/pytest/helpers.py

```diff
@@ -533,6 +533,212 @@ def shell_test_false():
+        replace_chars = ("-", " ")
+        name = self.name
+        for char in replace_chars:
+            name = name.replace(char, "_")
+        return name
+
+    def __attrs_post_init__(self):
+        self._laydown_files()
+
+    def _laydown_files(self):
+        if not self.srcdir.exists():
+            self.srcdir.mkdir()
+        setup_py = self.srcdir.joinpath("setup.py")
+        if not setup_py.exists():
+            setup_py.write_text(
+                textwrap.dedent(
+                    """\
+            import setuptools
+
+            if __name__ == "__main__":
+                setuptools.setup()
+            """
+                )
+            )
+        setup_cfg = self.srcdir.joinpath("setup.cfg")
+        if not setup_cfg.exists():
+            setup_cfg.write_text(
+                textwrap.dedent(
+                    """\
+            [metadata]
+            name = {0}
+            version = 1.0
+            description = Salt Extension Test
+            author = Pedro
+            author_email = pedro@algarvio.me
+            keywords = salt-extension
+            url = http://saltstack.com
+            license = Apache Software License 2.0
+            classifiers =
+                Programming Language :: Python
+                Programming Language :: Cython
+                Programming Language :: Python :: 3
+                Programming Language :: Python :: 3 :: Only
+                Development Status :: 4 - Beta
+                Intended Audience :: Developers
+                License :: OSI Approved :: Apache Software License
+            platforms = any
+
+            [options]
+            zip_safe = False
+            include_package_data = True
+            packages = find:
+            python_requires = >= 3.5
+            setup_requires =
+              wheel
+              setuptools>=50.3.2
+
+            [options.entry_points]
+            salt.loader=
+              module_dirs = {1}
+              runner_dirs = {1}.loader:get_runner_dirs
+              wheel_dirs = {1}.loader:get_new_style_entry_points
+            """.format(
+                        self.name, self.pkgname
+                    )
+                )
+            )
+
+        extension_package_dir = self.srcdir / self.pkgname
+        if not extension_package_dir.exists():
+            extension_package_dir.mkdir()
+            extension_package_dir.joinpath("__init__.py").write_text("")
+            extension_package_dir.joinpath("loader.py").write_text(
+                textwrap.dedent(
+                    """\
+            import pathlib
+
+            PKG_ROOT = pathlib.Path(__file__).resolve().parent
+
+            def get_module_dirs():
+                return [str(PKG_ROOT / "modules")]
+
+            def get_runner_dirs():
+                return [str(PKG_ROOT / "runners1"), str(PKG_ROOT / "runners2")]
+
+            def get_new_style_entry_points():
+                return {"wheel": [str(PKG_ROOT / "the_wheel_modules")]}
```

Comment:
**suggestion** similar to above: add some non-existent paths, or paths that have broken imports to ensure that our error reporting is :ok_hand:

Reply:
Will do on a separate PR